modules/currency.js

```diff
@@ -0,0 +1,220 @@
+
+import bidfactory from 'src/bidfactory';
+import { STATUS } from 'src/constants';
+import { ajax } from 'src/ajax';
+import * as utils from 'src/utils';
+import bidmanager from 'src/bidmanager';
+
+$$PREBID_GLOBAL$$.currency = setConfig;
+
+const DEFAULT_CURRENCY_RATE_URL = 'http://currency.prebid.org/latest.json';
+const CURRENCY_RATE_PRECISION = 4;
+
+var bidResponseQueue = [];
+var conversionCache = {};
+var currencyRatesLoaded = false;
+var adServerCurrency = 'USD';
+
+// Used as reference to the original bidmanager.addBidResponse
+var originalBidResponse;
+
+export var currencySupportEnabled = false;
+export var currencyRates = {};
+var currencyOverrides = {};
+
+export function setConfig(config) {
+  var url = DEFAULT_CURRENCY_RATE_URL;
+  if (typeof config.adServerCurrency === 'string') {
+    utils.logInfo('enabling currency support', arguments);
+
+    adServerCurrency = config.adServerCurrency;
+    if (config.conversionRateFile) {
+      utils.logInfo('currency using override conversionRateFile:', config.conversionRateFile);
+      url = config.conversionRateFile;
+    }
+    initCurrency(url);
+  } else {
+    // currency support is disabled, setting defaults
+    utils.logInfo('disabling currency support');
+    resetCurrency();
+  }
+  if (typeof config.currencyOverrides === 'object') {
+    currencyOverrides = config.currencyOverrides;
+  }
+}
+
+function initCurrency(url) {
+  conversionCache = {};
+  currencySupportEnabled = true;
+
+  if (!originalBidResponse) {
+    utils.logInfo('Installing addBidResponse decorator for currency module', arguments);
+
+    originalBidResponse = bidmanager.addBidResponse;
+    bidmanager.addBidResponse = addBidResponseDecorator(bidmanager.addBidResponse);
```

Comment:
> If we remove a method or change a signature on anything we export, they could reasonably claim that it's a breaking change to the library.

That's true but the same would be true if we created a plugin system.  I do think creating a standardized way to implement a plugin to Prebid.js core would be nice though; although a big undertaking, so I wonder if we should wait until we have a few more use cases other than currency?

In the current state I'm not sure how we could do this without this decorator.  I considered using the event system, but that would require mutating the bid objects, which seemed hacky; plus I don't think there would be any way using events to _defer_ the bid response (which is required if currency file hasn't loaded yet).

If there's a better way to do the currently provided functionality I'm open to hearing it.  Otherwise, maybe we could put a plugin system on the backlog for the future when we have more modules that require such features and we have a more formal insight into their needs?

It's possible the need for that plugin system is now though if we can't figure out a solution for this module and concurrent auctions.